### PR TITLE
dotCMS/core#22989 

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* setup Experiment Scheduling 

https://user-images.githubusercontent.com/31667212/216082504-f2cb3db4-5471-4131-81de-83174c6590a9.mov

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/services/dot-experiments.service.ts [hunk 1]
```diff
@@ -140,6 +146,22 @@ export class DotExperimentsService {
...
     }
 
+    /**
+     * Set scheduling to an experiment
+     * @param {string} experimentId
+     * @param {RangeOfDateAndTime | null} scheduling
+     * @returns Observable<DotExperiment>
+     * @memberof DotExperimentsService
+     */
+    setScheduling(
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/services/dot-experiments.service.ts:156]: Add the test of the service.

Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/shared/services/dot-experiments.service.ts:156]: done

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling-add/dot-experiments-configuration-scheduling-add.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    it('should close sidebar', () => {
+        spyOn(store, 'closeSidebar');
+        sidebar = spectator.query(Sidebar);
+        sidebar.hide();
+
+        expect(store.closeSidebar).toHaveBeenCalledTimes(1);
+    });
+
+    //TODO: Will be tested once defined the rules of edit.
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling-add/dot-experiments-configuration-scheduling-add.component.spec.ts:107]: Typically SonarQube cries about TODOs but here looks like no?

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling-add/dot-experiments-configuration-scheduling-add.component.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        CalendarModule
+    ],
+    templateUrl: './dot-experiments-configuration-scheduling-add.component.html',
+    styleUrls: ['./dot-experiments-configuration-scheduling-add.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DotExperimentsConfigurationSchedulingAddComponent implements OnInit {
+    form: FormGroup;
+    scheduling: RangeOfDateAndTime;
+    statusList = Status;
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling-add/dot-experiments-configuration-scheduling-add.component.ts:47]: You using this?

File: core-web/libs/dotcms-models/src/lib/dot-experiments.model.ts [hunk 4]
```diff
@@ -45,9 +45,9 @@ export interface GoalCondition {
     value: string;
 }
 
-interface RangeOfDateAndTime {
-    startDate: Date;
-    endDate: Date;
+export interface RangeOfDateAndTime {
+    startDate: number;
+    endDate: number;
```
Comment (fmontes) [inline core-web/libs/dotcms-models/src/lib/dot-experiments.model.ts:50]: We sure about this? I mean having it as a date allows more control no?

**Review by fmontes — CHANGES_REQUESTED** [review]
Code review done

File: core-web/libs/dotcms-models/src/lib/dot-experiments.model.ts (hunk 4 above)
Comment (hmoreras) [reply core-web/libs/dotcms-models/src/lib/dot-experiments.model.ts:50]: the endpoint currently return the date in miliseconds, that will be defined in other task related to the edit scheduling.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling-add/dot-experiments-configuration-scheduling-add.component.spec.ts (hunk 2 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling-add/dot-experiments-configuration-scheduling-add.component.spec.ts:107]: looks like, `0 Code Smells`

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling-add/dot-experiments-configuration-scheduling-add.component.html [hunk 5]
```diff
@@ -0,0 +1,49 @@
+<ng-container *ngIf="vm$ | async as vm">
+    <p-sidebar [(visible)]="vm.status.isOpen" (onHide)="closeSidebar()" dotSidebar>
+        <dot-sidebar-header
+            [actionButtonTpl]="actionHeaderBtnTpl"
+            dotTitle="{{ 'experiments.configure.scheduling.name' | dm }}"
```
[the remark this answers is not in the document]
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling-add/dot-experiments-configuration-scheduling-add.component.html:5]: new issue for that, https://github.com/dotcms/core/issues/24012

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling-add/dot-experiments-configuration-scheduling-add.component.ts (hunk 3 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling-add/dot-experiments-configuration-scheduling-add.component.ts:47]: yes in the template, rename variable to `stepStatus`

**Review by oidacra — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**